### PR TITLE
EVG-2404: attach artifacts command

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: command/attach_artifacts.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+	c.Files, err = util.BuildFileList(conf.WorkDir, c.Files...)
+	if err != nil {
+		err = errors.Wrap(err, "problem building wildcard paths")
+		logger.Task().Error(err)
+		return err
+	}
+
+	if len(c.Files) < 1 {
```
[the remark this answers is not in the document]
Comment (tychoish) [reply command/attach_artifacts.go]: right but we're reassigning c.Files, so after the gitignore file list parsing, we might end up with less than we started.

File: command/attach_artifacts.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		err = errors.New("expanded file specification had no items")
+		logger.Execution().Error(err)
```
[the remark this answers is not in the document]
Comment (tychoish) [reply command/attach_artifacts.go]: done

File: command/attach_artifacts.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	var segment []*artifact.File
+	for idx := range c.Files {
+		segment, err = readArtifactsFile(conf.WorkDir, c.Files[idx])
+		if err != nil {
+			catcher.Add(err)
+			continue
+		}
+
+		if segment == nil {
+			continue
```
[the remark this answers is not in the document]
Comment (tychoish) [reply command/attach_artifacts.go]: done

File: command/attach_artifacts.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+		}
+
+		files = append(files, segment...)
+	}
+
+	if catcher.HasErrors() {
+		err = errors.Wrap(catcher.Resolve(), "encountered errors reading artifact json files")
+		logger.Execution().Error(err)
```
[the remark this answers is not in the document]
Comment (tychoish) [reply command/attach_artifacts.go]: done

File: command/attach_artifacts.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bsamek) [reply command/attach_artifacts.go]: I'm not saying that this is duplication. I'm saying you should check length `== 0`, not length `< 1`.

File: command/attach_artifacts.go [hunk 5]
```diff
@@ -91,7 +97,7 @@ func (c *attachArtifacts) Execute(ctx context.Context,
 		return errors.Wrap(err, "attach artifacts failed")
 	}
 
-	logger.Execution().Infof("'$s' attached %d resources to task", c.Name(), len(files))
+	logger.Task().Infof("$s attached %d resources to task", c.Name(), len(files))
```
Comment (bsamek) [inline command/attach_artifacts.go]: `%s`

**Comment by tychoish** [comment]
Updated

**Pull request merged**